[TensorExpr] Add lowerings for quantized ops: cat, mul, conv1d, relu.

ZolotukhinM opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #66522
* #66847
* #66515
* #66514
* __->__ #69055
* #68857
* #68856
* #68855
* #68854

Differential Revision: [D32710325](https://our.internmc.facebook.com/intern/diff/D32710325)

Commit:
[TensorExpr] Add lowerings for quantized ops: cat, mul, conv1d, relu.

[ghstack-poisoned]

ZolotukhinM commented:
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32710325).

Commit:
Update on "[TensorExpr] Add lowerings for quantized ops: cat, mul, conv1d, relu."


Differential Revision: [D32710325](https://our.internmc.facebook.com/intern/diff/D32710325)

[ghstack-poisoned]

ZolotukhinM commented:
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32710325).

navahgar commented on torch/csrc/jit/tensorexpr/external_functions.cpp:
```diff
@@ -479,6 +506,34 @@ void nnc_aten_quantized_mul_scalar(
+      a_qscale,
+      a_qzero,
+      toQIntType(a_qdtype));
+  auto r = at::relu(qa);
```
Should this be calling `quantized_relu` function instead? (like the `nnc_aten_quantized_cat` case below)

ZolotukhinM replied:
I'm not sure, but it does produce correct results :) I suspect it dispatches to `quantized::relu` under the hood when `qa` is a QTensor.

Commit:
Update on "[TensorExpr] Add lowerings for quantized ops: cat, mul, conv1d, relu."


Differential Revision: [D32710325](https://our.internmc.facebook.com/intern/diff/D32710325)

[ghstack-poisoned]

ZolotukhinM commented:
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32710325).

navahgar reviewed (approved):
LGTM

Commit:
Update on "[TensorExpr] Add lowerings for quantized ops: cat, mul, conv1d, relu."


Differential Revision: [D32710325](https://our.internmc.facebook.com/intern/diff/D32710325)

[ghstack-poisoned]

ZolotukhinM commented:
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32710325).

Commit:
Update on "[TensorExpr] Add lowerings for quantized ops: cat, mul, conv1d, relu."


Differential Revision: [D32710325](https://our.internmc.facebook.com/intern/diff/D32710325)

[ghstack-poisoned]

ZolotukhinM commented:
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32710325).

Commit:
Update on "[TensorExpr] Add lowerings for quantized ops: cat, mul, conv1d, relu."


Differential Revision: [D32710325](https://our.internmc.facebook.com/intern/diff/D32710325)

[ghstack-poisoned]

ZolotukhinM commented:
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32710325).
